### ISSUE TITLE
ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './dist/ts-jest-transformer' is not defined by "exports".

### ISSUE THREAD

**Issue by ferreria23** [thymikee/jest-preset-angular#941]
## To Reproduce

Just upgrade from jest-present-angular 8.3.1 to 9.0.1 and now I cannot run tests.

## Expected behavior

The tests should run.
## Error log:

```
# content of error stacktrace :
   Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './dist/ts-jest-transformer' is not defined by "exports" in /home/x/
y/z/node_modules/ts-jest/package.json

      at Object.<anonymous> (node_modules/jest-preset-angular/build/ng-jest-transformer.js:4:31)
```

## envinfo

```
System:
    OS: WSL2 - Debian

Npm packages:
    jest: 27.0.3
    jest-preset-angular: 9.0.1
    typescript: 4.0.2
```

**Comment by ahnpnl** [comment]
Hi, would you please try to clear jest cache?

If that doesn’t work, pls provide your repo :)

**Comment by ferreria23** [comment]
Hello @ahnpnl ,

I removed the cache from jest but still giving the same error.

Saddly the repo is private and I cannot provide it.

**Comment by ahnpnl** [comment]
would you please check what is the version of `ts-jest`?

If it is a private repo, would you please make a separate dummy repo to reproduce the issue?

**Comment by ferreria23** [comment]
ts-jest version is 27.0.2.

**Comment by ahnpnl** [comment]
Does removing `node_modules` and reinstall deps help?

Are you using nrwl nx or using angular-builder/jest?

I have a sample repo here https://github.com/ahnpnl/jest-angular but I cannot reproduce your problem.

**Comment by ahnpnl** [comment]
Are you using `yarn` or `npm`? How did you run the upgrade?

**Comment by j0nm1** [comment]
I'm using npm 6.14.4 and updated with `npm i -D jest-preset-angular@9.0.1`

**Comment by ahnpnl** [comment]
Do you have a repo which can be always producible?

**Comment by banjankri** [comment]
Had the same issue - turned out to be node version problem - had run with Node v14.6. Hope that helps somebody.

**Comment by ahnpnl** [comment]
I suppose Node 12.22.1 will also work. `jest-preset-angular` CI use Node 12.22.1 and Node 14.16.x

**Comment by ferreria23** [comment]
My Node version is 12.18.0.

**Comment by ahnpnl** [comment]
@ferreria23 would you please try with Node 12.22.1? I see from NodeJs doc mentioned https://nodejs.org/api/packages.html#packages_subpath_patterns was added in **12.20.0**

**Comment by j0nm1** [comment]
I also had node 12.18.0 installed. After upgrading to 14.17.0 (latest lts) it works fine. Thank you for your help!

**Comment by ahnpnl** [comment]
I will bring this up to `ts-jest`, seem like node at 12.20 is required.

**Comment by ahnpnl** [comment]
Confirmed that Node < 12.20.0 will get this issue. I could reproduce by switching GitHub action to Node 12.18.0 https://github.com/ahnpnl/jest-preset-angular/runs/2736935738?check_suite_focus=true

**Comment by Random90** [comment]
I have same issue, while updating from "jest-preset-angular": "7.1.1" to "jest-preset-angular": "^9.0.1".
Node version is v14.5.0 though.

**Comment by ahnpnl** [comment]
Node 12.20.0+ or 14.13.0+, see https://nodejs.org/api/packages.html#packages_subpath_patterns

**Comment by thovo** [comment]
I have the same issue on my windows with NodeJS 12.18.3 for a new Angular project, after upgrade to the latest node version, it worked just fine.

**Comment by ahnpnl** [comment]
fix in **9.0.2**
